test/extended/util/disruption/disruption.go

```diff
@@ -122,45 +117,68 @@ func runChaosmonkey(
 			TestData:        testData,
 			framework:       f,
 			test:            t,
-			testReport:      testCase,
+			testName:        displayName,
+			className:       "disruption_tests",
 			testSuiteReport: testSuite,
 		}
 		cm.Register(cma.Test)
 	}
 
 	start := time.Now()
 	defer func() {
-		testSuite.Update()
-		testSuite.Time = time.Since(start).Seconds()
-
-		// if the test fails and all failures are described as "Flake", create a second
-		// test case that is listed as success so the test is properly marked as flaky
-		for _, testCase := range testSuite.TestCases {
-			allFlakes := len(testCase.Failures) > 0 && len(testCase.Errors) == 0 && len(testCase.Skipped) == 0
-			for _, failure := range testCase.Failures {
-				if failure.Type == "Flake" {
-					failure.Type = "Failure"
-				} else {
-					allFlakes = false
-				}
+
+		// Calculate NumFailed and NumSkipped
```

Comment:
These populate the counts in the top level <testsuite>, has to be tracked manually and I found calculating here safer than trying to increment everywhere we add a test case.